src/adapters/MongoStorageAdapter.js

```diff
@@ -113,9 +132,47 @@ const MongoStorageAdapter = function (uuid) {
 			cb(err, { status: "failed" });
 		});
 	};
-};
 
+	/**
+	 * Get the prefix used to filter keys for particular topics and key prefixes.
+	 *
+	 * @param {object} params
+	 * @param {string} params.topic The topic
+	 * @param {string} params.keyPrefix The key prefix (optional).
+	 * @private
+	 */
+	getKeyPreface = function(params) {
+		const keyPrefix = "keyPrefix" in params ? params.keyPrefix : "";
+		const preface = `${this.getUserPreface()}:${params.topic}:${keyPrefix}`;
+
+		return preface;
+	};
+
+	/**
+	 * Get prefix for all the users stored data.
+	 * @private
+	 */
+	getUserPreface = function() {
+		const preface = `${this.baseName}:${this.userName}`;
```

Comment:
```suggestion
		const preface = `${this.baseName}:${this.userName}:`;
```
Ensure userName arg is not greedy